src/node_buffer.cc

```diff
@@ -219,7 +219,10 @@ MaybeLocal<Object> New(Isolate* isolate,
   CHECK(actual <= length);
 
   if (actual < length) {
-    data = static_cast<char*>(realloc(data, actual));
+    char* new_data = static_cast<char*>(realloc(data, actual));
```

Comment:
Sorry, missed one thing.
This PR changes valid behavior to invalid.
Without this PR, if `realloc` fails, the program just asserts.
With this PR, if `realloc` fails, the program continues like nothing bad has happened. You actually broke the assert.